lib/pf/UnifiedApi/Controller/Config.pm

```diff
@@ -1300,10 +1305,39 @@ sub bulk_import {
         return $self->render(json => $data, status => $status);
     }
 
+    if ($data->{async}) {
+        my $task_id = $self->task_id;
+        my $subprocess = Mojo::IOLoop->subprocess;
+        $subprocess->run(
+            sub {
+                my ($subprocess) = @_;
+                my $updater = pf::pfqueue::status_updater::redis->new( connection => consumer_redis_client(), task_id => $task_id );
+                $updater->start;
+                my $results = $self->do_bulk_import($data);
+                $updater->completed({items => $results});
+                return;
+            },
+            sub { } # Do nothing
+        );
+
+        return $self->render( json => {status => 202, task_id => $task_id }, status => 202);
+    }
+
+    my $results = $self->do_bulk_import($data);
```

Comment:
An else statement would make things clearer that this is the synchronous import that will only be called if async is false